core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -109,6 +111,9 @@ class KryoSerializer(conf: SparkConf)
     kryo.register(classOf[SerializableJobConf], new KryoJavaSerializer())
     kryo.register(classOf[HttpBroadcast[_]], new KryoJavaSerializer())
     kryo.register(classOf[PythonBroadcast], new KryoJavaSerializer())
+    kryo.register(classOf[TaskMetrics], new KryoJavaSerializer())
+    kryo.register(classOf[DirectTaskResult[_]], new KryoJavaSerializer())
+    kryo.register(classOf[IndirectTaskResult[_]], new KryoJavaSerializer())
```

Comment:
FYI, each of these calls is actually somewhat expensive. We create thousands of new Kryo serializer instances because we don't reuse them and so we end up calling `register` here many times. There is an outstanding issue [SPARK-12473](https://issues.apache.org/jira/browse/SPARK-12473) to resolve this, but until that is fixed this commit will likely cause minor performance regressions.